core/base/channel_test.go

```diff
@@ -22,6 +22,9 @@ func (s *ChannelSuite) TestParse(c *gc.C) {
 	ch, err = ParseChannel("22.04/edge")
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(ch, jc.DeepEquals, Channel{Track: "22.04", Risk: "edge"})
+	ch, err = ParseChannel("all/stable")
```

Comment:
"all" with a risk isn't possible in real life, or shouldn't be? It'd only be "all". This is a hold over from charms imported from charmstore.

Reply:
Interesting. The reason this is here is because of this `if` block:
https://github.com/juju/juju/blob/3.3/core/charm/repository/charmhub.go#L1014

If this is a hangover from charmstore, however, I will drop